### PR TITLE
Partner Portal: Update partner key handling to match changes in the API

### PR DESCRIPTION
**Pull request by atanas-dev**
Context:
* Project thread: pbtFPC-Um-p2
* i4 designs: pbtFPC-103-p2
* D55528-code necessitates this change as it changes the API endpoint and response format.

#### Changes proposed in this Pull Request

* Update partner key handling in the Partner Portal to match the upcoming changes in the Licensing API (/partnrers -> /partner endpoint and response format).

#### Testing instructions

* Patch and sandbox D55528-code unless it is already deployed.
* Apply this PR to your local environment.
* Open up http://jetpack.cloud.localhost:3000/partner-portal/partner-key
  * If you have a partner with a valid active key you will be redirected to a welcome screen.
  * If not, you will be shown a very rough error screen that you do not have a partner account.

Note: Changeset looks bigger due to plural to singular renames.

### CODE REVIEW

File: client/lib/wpcom-undocumented/lib/undocumented.js
Comment (ebinnion) [inline client/lib/wpcom-undocumented/lib/undocumented.js]: Instead of filtering in JS, could we just add the option to only get active keys in the API? Alternatively, could/should we only return active keys in the response by default?

File: client/jetpack-cloud/sections/partner-portal/select-partner-key/index.tsx
Comment (ebinnion) [inline client/jetpack-cloud/sections/partner-portal/select-partner-key/index.tsx:47]: This is interesting. Why do we have an empty react element here instead of using something like a `<div>`?

File: client/state/partner-portal/reducer/partner.ts
Comment (ebinnion) [inline client/state/partner-portal/reducer/partner.ts:55]: Similar to my above comment, should we move this logic into the API instead of the client?

**Review by ebinnion — APPROVED** [review]
I left a couple of non-blocking comments. This tests well functionally.

File: client/jetpack-cloud/sections/partner-portal/select-partner-key/index.tsx
Comment (atanas-dev) [reply client/jetpack-cloud/sections/partner-portal/select-partner-key/index.tsx:47]: A div is unnecessary in this case. The Query component renders nothing and I wanted to hoist it at the top for visibility hence the empty (fragment) component. Alternatively, I can nest it at the top of `<Main>` and drop the `<></>`.

File: client/lib/wpcom-undocumented/lib/undocumented.js
Comment (atanas-dev) [reply client/lib/wpcom-undocumented/lib/undocumented.js]: Done: D55774-code
~Will merge as-is and remove after D55774-code has landed.~

File: client/state/partner-portal/reducer/partner.ts
Comment (atanas-dev) [reply client/state/partner-portal/reducer/partner.ts:55]: I think it's fine if the API still outputs a partner even if that partner does not have any valid keys as I think it's an important distinction to have between a normal WPCOM user and a WPCOM user who has a partner account albeit with no keys.

**Pull request merged**
